src/DownloadManager.ts

```diff
@@ -0,0 +1,243 @@
+import { logger } from './Logger';
+
+import cliProgress from 'cli-progress';
+import WebSocket from 'ws';
+
+
+export class DownloadManager {
+    private webSocket: WebSocket;
+    // TODO: there's a "not a tty" mode for progresBar
+    // NOTE: is there a way to fix the ETA? Can't get size nor ETA from aria that I can see
+    // we initialize this for each download
+    private progresBar!: cliProgress.Bar;
+    private completed: number;
+    private queue: Set<string>;
+    private index: number;
+
+    public constructor(port: number) {
```

Comment:
Which port is this, 6800?

```
--rpc-listen-port=<PORT>
Specify a port number for JSON-RPC/XML-RPC server to listen to. Possible Values: 1024 -65535
Default: 6800
```

Here's a better plan, let's check if that port if available and noone else listens on it else let's increment port+1 and try to listen again.

Ref: https://stackoverflow.com/a/29872303/4148708

We could use aria's binary to probe the listen port IF it throws a meaningful exit code back should the port be taken.

Reply:
done..ish